tests/test-sessions/test-infinispan-sessions/src/test/java/org/eclipse/jetty/server/session/InfinispanSessionDataStoreTest.java

```diff
@@ -164,30 +166,26 @@ public boolean checkSessionPersisted(SessionData data) throws Exception
     @Test
     public void testQuery() throws Exception
     {
-        Cache<String, SessionData> cache = _testSupport.getCache();
+        InfinispanSessionData sd1 = new InfinispanSessionData("sd1", "", "", 0, 0, 0, 1000);
+        sd1.setLastNode("fred1");
+        _testSupport.getCache().put("session1", sd1);
 
-        SessionData sd1 = new SessionData("sd1", "", "", 0, 0, 0, 0);
-        SessionData sd2 = new SessionData("sd2", "", "", 0, 0, 0, 1000);
-        sd2.setExpiry(100L); //long ago
-        SessionData sd3 = new SessionData("sd3", "", "", 0, 0, 0, 0);
+        InfinispanSessionData sd2 = new InfinispanSessionData("sd2", "", "", 0, 0, 0, 2000);
+        sd2.setLastNode("fred2");
+        _testSupport.getCache().put("session2", sd2);
 
-        cache.put("session1", sd1);
-        cache.put("session2", sd2);
-        cache.put("session3", sd3);
+        InfinispanSessionData sd3 = new InfinispanSessionData("sd3", "", "", 0, 0, 0, 3000);
+        sd3.setLastNode("fred3");
+        _testSupport.getCache().put("session3", sd3);
 
-        QueryFactory qf = Search.getQueryFactory(cache);
-        Query q = qf.from(SessionData.class).select("id").having("expiry").lte(System.currentTimeMillis()).and().having("expiry").gt(0).toBuilder().build();
+        QueryFactory qf = Search.getQueryFactory(_testSupport.getCache());
 
-        List<Object[]> list = q.list();
-
-        List<String> ids = new ArrayList<>();
-        for (Object[] sl : list)
+        for (int i = 0; i <= 3; i++)
         {
-            ids.add((String)sl[0]);
+            long now = System.currentTimeMillis();
+            Query q = qf.from(InfinispanSessionData.class).having("expiry").lt(now).build();
```

Comment:
Also, why did you change the query? The query did the same search as the InfinispanSessionDataStore implementation.